File: src/lib/src/main/java/work/lclpnet/ap2/api/game/team/TeamKey.java
package work.lclpnet.ap2.api.game.team;

import net.minecraft.ChatFormatting;
import work.lclpnet.kibu.translate.Translations;
import work.lclpnet.kibu.translate.text.TranslatedText;

public sealed interface TeamKey permits DyeTeamKey {

    String id();

    int color();

    ChatFormatting formatting();

    default String getTranslationKey() {
        return "ap2.team." + id();
    }

    default TranslatedText getDisplayName(Translations translations) {
        return translations.translateText(getTranslationKey())
                .styled(style -> style.withColor(color()));
    }
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/game/team/TeamKey.kt
package work.lclpnet.ap2.game.team

import net.minecraft.world.scores.TeamColor
import work.lclpnet.kibu.translate.Translations
import work.lclpnet.kibu.translate.text.TranslatedText

interface TeamKey {

    val id: String

    val color: Int

    val teamColor: TeamColor

    val translationKey: String
        get() = "ap2.team.$id"

    fun getDisplayName(translations: Translations): TranslatedText =
        translations.translateText(this.translationKey)
            .withColor(color)
}
